compiler/src/iree/compiler/Dialect/Stream/Transforms/test/encode_device_tensors_packing.mlir

```diff
@@ -0,0 +1,119 @@
+// RUN: iree-opt --split-input-file --iree-stream-encode-device-tensors --verify-diagnostics %s | FileCheck %s
+
+// Check that i3 load is expanded right now.
+
+// CHECK-LABEL: @subspanLoadI3
+stream.executable private @subspanLoadI3 {
+  stream.executable.export public @dispatch
+  builtin.module {
+    func.func @dispatch(%arg0: !stream.binding) {
+      %c0 = arith.constant 0 : index
+      // CHECK: %[[BINDING:.+]] = stream.binding.subspan {{.+}} -> !flow.dispatch.tensor<readonly:tensor<4xi8>>
+      %binding = stream.binding.subspan %arg0[%c0] : !stream.binding -> !flow.dispatch.tensor<readonly:tensor<4xi3>>
+      // CHECK: %[[TILE_I8:.+]] = flow.dispatch.tensor.load %[[BINDING]], offsets = [0], sizes = [4], strides = [1] : !flow.dispatch.tensor<readonly:tensor<4xi8>> -> tensor<?xi8>
+      // CHECK: %[[TILE_I3:.+]] = arith.trunci %[[TILE_I8]] : tensor<?xi8> to tensor<?xi3>
+      %tile = flow.dispatch.tensor.load %binding, offsets = [0], sizes = [4], strides = [1] : !flow.dispatch.tensor<readonly:tensor<4xi3>> -> tensor<?xi3>
+      // CHECK: util.optimization_barrier %[[TILE_I3]] : tensor<?xi3>
+      util.optimization_barrier %tile : tensor<?xi3>
+      return
+    }
+  }
+}
+
+// -----
+
+// Check that i3 store is expanded right now.
```

Comment:
try i13 instead to ensure the code handles non-byte-aligned values that are over 1 byte

Reply:
We already have tests for i33 in the normal `encode_device_tensors.mlir` file:

https://github.com/openxla/iree/blob/4552fda2fab0290f0172f6b30acf5f010138015d/compiler/src/iree/compiler/Dialect/Stream/Transforms/test/encode_device_tensors.mlir#L70-L86

This `*_packing.mlir` file is newly added for for sub-byte packing cases.

Reply:
Actually that's for load. I just added another test case for store.